pkgs/games/dwarf-fortress/wrapper/dwarf-fortress-init.in

```diff
@@ -30,16 +38,19 @@ forcecopy_path() {
 Using $DF_DIR as Dwarf Fortress overlay directory.
 If you do any changes in it, don't forget to clean it when updating the game version!
 We try to detect changes based on data directories being symbolic links -- keep this in mind.
-
 EOF
 
 cd "$env_dir"
-for i in data/init/* data/!(init|index|announcement) raw; do
-  update_path "$i"
+for path in libfmod_plugin.so libsdl_mixer_plugin.so data/init/* data/!(init|index|announcement) raw libs dwarfort; do
+  if [ -e "$path" ]; then
+    update_path "$path"
+  else
+    cleanup_path "$path"
+  fi
```

Comment:
```
error: builder for '/nix/store/1hzq17s5wqdr843iwbpn861ndfvq48h9-dwarf-fortress-config.drv' failed with exit code 1;
       last 1 log lines:
       > cp: cannot stat '/nix/store/np8f97rfbgwg465kpk2277r3azshig18-dwarf-fortress-base-env-50.12/data/init/init.txt': No such file or directory
       For full logs, run 'nix log /nix/store/1hzq17s5wqdr843iwbpn861ndfvq48h9-dwarf-fortress-config.drv'.
error: 1 dependencies of derivation '/nix/store/nv9zlhlk5n30mnbrpaysy4jymj3y6p3k-dwarf-fortress-env-50.12.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4hi7q2aicspyhyzm7pigd8yypgfzw8jb-dwarf-fortress-init.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vx7mzqh435j9sfk583l45xkm7ybfr82w-dwarf-fortress-50.12.drv' failed to build
```

Reply:
```
$ ls /nix/store/np8f97rfbgwg465kpk2277r3azshig18-dwarf-fortress-base-env-50.12/data/init/*
/nix/store/np8f97rfbgwg465kpk2277r3azshig18-dwarf-fortress-base-env-50.12/data/init/announcements.txt
/nix/store/np8f97rfbgwg465kpk2277r3azshig18-dwarf-fortress-base-env-50.12/data/init/arena.txt
/nix/store/np8f97rfbgwg465kpk2277r3azshig18-dwarf-fortress-base-env-50.12/data/init/colors.txt
/nix/store/np8f97rfbgwg465kpk2277r3azshig18-dwarf-fortress-base-env-50.12/data/init/d_init_default.txt
/nix/store/np8f97rfbgwg465kpk2277r3azshig18-dwarf-fortress-base-env-50.12/data/init/init_default.txt
/nix/store/np8f97rfbgwg465kpk2277r3azshig18-dwarf-fortress-base-env-50.12/data/init/interface.txt
```

Reply:
What derivation are you building? I think I see what the issue is:

```
  '' + forEach settings' (file: kv: ''              
    file=data/init/${lib.escapeShellArg file}.txt
    if [ -f "$file" ]; then # added
      cp ${baseEnv}/"$file" "$out/$file"
    fi     # added
```

But I'm curious where this is from (it's not the runtime of the wrapper, it's the build time...)

Reply:
Give it a shot now, I think I reproduced a case similar to yours and was able to fix it.

Reply:
Still getting this with just `nix-build -A dwarf-fortress`.

Reply:
The default:
```
Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/300402

1 package failed to build:
dwarf-fortress
```

Reply:
Oooh, I see what's up. That derivation is trying to build a bunch of different versions.

Reply:
Give it a shot now.